pymatgen/io/vasp/inputs.py

```diff
@@ -2280,6 +2283,43 @@ def __repr__(self) -> str:
         return f"{cls_name}({symbol=}, {functional=}, {TITEL=}, {VRHFIN=}, {n_valence_elec=:.0f})"
 
 
+def _gen_potcar_summary_stats():
```

Comment:
This is fine here, but noting there is also a top-level `dev_scripts` folder (largely undocumented). On balance it's probably better here for discoverability for anyone curious how this file was generated.

Minor comments: the bare `return` is unnecessary, and `monty.serialization.dumpfn` can be used directly with `dumpfn(new_summary_stats, "filename.json.gz")` which is a convenience function we use a lot.

Reply:
Thanks! Much better way of doing this. For placement, you and @janosh probably have a better sense of where this belongs. `_gen_potcar_summary_stats` is a "destructive" function, in that it overwrites the file that's distributed with PMG. `dev_scripts` might then be a better place for it